arch/risc-v/src/mpfs/mpfs_ihc.c

```diff
@@ -965,11 +965,11 @@ mpfs_rptun_get_resource(struct rptun_dev_s *dev)
       rsc->rpmsg_vring0.align       = VRING_ALIGN;
       rsc->rpmsg_vring0.num         = VRING_NR;
       rsc->rpmsg_vring0.da          = VRING0_DESCRIPTORS;
-      rsc->rpmsg_vring0.notifyid    = 0;
+      rsc->rpmsg_vring0.notifyid    = 1;
       rsc->rpmsg_vring1.align       = VRING_ALIGN;
       rsc->rpmsg_vring1.num         = VRING_NR;
       rsc->rpmsg_vring1.da          = VRING1_DESCRIPTORS;
-      rsc->rpmsg_vring0.notifyid    = 1;
+      rsc->rpmsg_vring0.notifyid    = 0;
```

Comment:
Should be rsc->rpmsg_vring1.notifyid    = 0;  (not vring0, apparent typo).. need to check

Reply:
it's better to initialize notifyid to RSC_NOTIFY_ID_ANY, so the framework will allocate the unique id for you automatically:
https://github.com/OpenAMP/open-amp/blob/main/lib/remoteproc/rsc_table_parser.c#L131-L136

Reply:
I updated sim's implementation to reflect this fact: https://github.com/apache/incubator-nuttx/pull/6903

Reply:
@xiaoxiang781216 Thanks for the input ,I'll try how it goes

Reply:
Looks like it might create ids 1 & 2, but the Linux side expects fixed 0 and 1. And only id 0 is sent to Linux kernel.

Reply:
> Looks like it might create ids 1 & 2,

Because the first id is allocated to fw_rsc_vdev:
https://github.com/OpenAMP/open-amp/blob/main/lib/include/openamp/remoteproc.h#L296
and next two to fw_rsc_vdev_vring:
https://github.com/OpenAMP/open-amp/blob/main/lib/include/openamp/remoteproc.h#L253

> but the Linux side expects fixed 0 and 1. And only id 0 is sent to Linux kernel.

I have long time not watch the Linux change, but I never hit this limitation before when I am working with remoteproc/rpmsg several years ago.

Reply:
Yes, this restriction comes from Microchip sw, not the rpmsg itself.